src/fides/api/service/connectors/saas_connector.py

```diff
@@ -192,7 +192,7 @@ def retrieve_data(
         # 1) If a collection can be retrieved by using different identities such as email or phone number
         # 2) The complete set of results for a collection is made up of subsets. For example, to retrieve all tickets
         #    we must change a 'status' query param from 'active' to 'pending' and finally 'closed'
-        read_requests: List[SaaSRequest] = query_config.get_read_requests()
+        read_requests: List[SaaSRequest] = query_config.get_read_requests_by_identity()
```

Comment:
The issue here is that we would return _all_ read requests, instead of just the ones for the provided identities. Bringing this back to filter the read requests but still allow more than one identity.